runtime/IO/File.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <AK/Array.h>
+#include <AK/Error.h>
+#include <AK/RefCounted.h>
+#include <AK/String.h>
+#include <stdio.h>
+
+class File final : public RefCounted<File> {
+public:
+    static ErrorOr<NonnullRefPtr<File>> open_for_reading(String path);
+    static ErrorOr<NonnullRefPtr<File>> open_for_writing(String path);
```

Comment:
can these be const references, given that the parameters in the jakt declaration are not `mutable`?

Reply:
Hmm, interesting question. Of course they *could* because you and I know that it's safe.

However, jakt has no concept of "const references", it sees `String` as a value type. For that reason, it feels natural to pass `String` rather than `String const&`.

Reply:
fair enough!